examples/step-69/doc/intro.dox

```diff
@@ -0,0 +1,345 @@
+@f}
+
+holds true, where $|\!|\!| \cdot |\!|\!| $ could represent the
+$L^2(\Omega)$-norm or, more generally, some discrete (possibly mesh
+dependent) energy-norm. Variational discretizations of hyperbolic
+conservation laws have been very popular since the mid eighties, in
+particular combined with SUPG-type stabilization and/or upwinding
+techniques (see the early work of @cite Brooks1982 and @cite Johnson1986). They
+have proven to be some of the best approaches for simulations in the subsonic
+shockless regime and similarly benign regimes.
+
+However, in the transonic and supersonic regime, and shock-hydrodynamics
+applications the use of variational schemes might be questionable. In fact,
+at the time of this writing, most shock-hydrodynamics codes are still
+firmly grounded on finite volumes methods. The main reason for failure of
+variational schemes in such extreme regimes is the lack of pointwise
+stability. This stems from the fact that <i>a priori</i> bounds on
+integrated quantities (e.g. integrals of moments) have in general no
+implications on pointwise properties of the solution. While some of these
+problems might be alleviated by the (perpetual) chase of the right shock
+capturing scheme, finite difference-like and finite volume schemes still
+have an edge in many regards.
+
+In this tutorial step we therefore depart from variational schemes. We will
+present a completely algebraic formulation (with the flavor of a
+collocation-type scheme) that preserves constraints pointwise, i.e.,
+
+@f{align*}
+  \textbf{u}_h(\mathbf{x}_i,t) \in \mathcal{B}
+  \;\text{at every node}\;\mathbf{x}_i\;\text{of the mesh}.
+@f}
+
+Contrary to finite difference/volume schemes, the scheme implemented in
+this step maximizes the use of finite element software infrastructure,
+works in any mesh, in any space dimension, and is theoretically guaranteed
+to always work, all the time, no exception. This illustrates that deal.ii
+can be used far beyond the context of variational schemes in Hilbert spaces
+and that a large number of classes, modules and namespaces from deal.ii can
+be adapted for such purpose.
+
+
+<h3>Description of the scheme </h3>
+
+Let $\mathbb{V}_h$ be scalar-valued finite dimensional space spanned by a
+basis $\{\phi_i\}_{i \in \mathcal{V}}$ where: $\phi_i:\Omega \rightarrow
+\mathbb{R}$ and $\mathcal{V}$ is the set of all indices (nonnegative
+integers) identifying each scalar Degree of Freedom (DOF) in the mesh.
+Therefore a scalar finite element functiona $u_h \in \mathbb{V}_h$ it can
+be written as $u_h = \sum_{i \in \mathcal{V}} U_i \phi_i$ with $U_i \in
+\mathbb{R}$. We introduce the notation for vector-valued approximation
+spaces $\pmb{\mathbb{V}}_h := \{\mathbb{V}_h\}^{d+2}$. Let $\mathbf{u}_h
+\in \pmb{\mathbb{V}}_h$, then it can be written as $\mathbf{u}_h = \sum_{i
+\in \mathcal{V}} \mathbf{U}_i \phi_i$ where $\mathbf{U}_i \in
+\mathbb{R}^{d+2}$ and $\phi_i$ is a scalar-valued shape function.
```

Comment:
In order to be compatible with the literature we cite (and almost all our notes) we have to retain this notation.

Reply:
The representation of vector valued shape functions (say FESystem) is very much appropriate for mixed  formulations (and other systems) where different components might use a different finite element spaces: the block-matrices that you compute describe interactions between dofs. For the methods illustrated in this tutorial the theory that exists (and possibly the only one it will ever exist) is one that requires every component to use the same scalar-valued finite element space. The interactions described by the scheme are between nodes not between dofs (this is a fundamental difference). The notation being used was deliberately chosen in order to emphasize this aspect of the scheme. Using the usual notation of vector valued shape functions would be somewhat misleading.

Reply:
I see. Thanks for the clarification -- maybe a sentence to this effect explaining the design choice would make sense?

Reply:
I'll add that the choice of making `U_i` scalar makes sense in the context of schemes that build matrices (or where one thinks of the operator as matrices). But you're not doing this: You're just looping over nodes in the mesh and updating values there. So I can see how that makes sense. The only tricky part is how you stick the components of these tensors into solution vectors for graphical output -- there needs to be some kind of mapping from your tensor-valued `U_i` to vector components.